contracts/src/structs/CrossNet.sol

```diff
@@ -104,3 +104,12 @@ struct ReceiptMsg {
     // TODO: In the future we may include here events and other
     // feedback information.
 }
+
+struct ApplyMsgRet {
+    /// The ret bytes from message execution
+    bytes ret;
+    /// Whether the execution is successful
+    bool success;
+    /// The cross net message executed
+    IpcEnvelope crossMsg;
```

Comment:
Is this returned so as not to have to clone the message when calling this?

Reply:
reason for this is that there are inplace modification to the crossmsg, so I cannot just use the original one. Maybe I should add a comment for this.